From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Sat, 21 Mar 2026 00:21:21 +0100
Subject: camera: Use permission ID to request pipewire to open the stream

Pipewire uses the permission portal to check if the application can
access by looking up the permission IDs, so we need to pass the
permission ID to it

Origin: https://github.com/flatpak/xdg-desktop-portal/pull/1764
---
 src/camera.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/camera.c b/src/camera.c
index 0e9f9e4..e19369d 100644
--- a/src/camera.c
+++ b/src/camera.c
@@ -225,7 +225,7 @@ handle_access_camera (XdpDbusCamera *object,
 }
 
 static PipeWireRemote *
-open_pipewire_camera_remote (const char *app_id,
+open_pipewire_camera_remote (const char *permissions_id,
                              GError **error)
 {
   PipeWireRemote *remote;
@@ -233,7 +233,7 @@ open_pipewire_camera_remote (const char *app_id,
   struct pw_properties *pipewire_properties;
 
   pipewire_properties =
-    pw_properties_new ("pipewire.access.portal.app_id", app_id,
+    pw_properties_new ("pipewire.access.portal.app_id", permissions_id,
                        "pipewire.access.portal.media_roles", "Camera",
                        NULL);
   remote = pipewire_remote_new_sync (pipewire_properties,
@@ -297,7 +297,7 @@ handle_open_pipewire_remote (XdpDbusCamera *object,
       return G_DBUS_METHOD_INVOCATION_HANDLED;
     }
 
-  remote = open_pipewire_camera_remote (xdp_app_info_get_id (app_info), &error);
+  remote = open_pipewire_camera_remote (xdp_app_info_get_permissions_id (app_info), &error);
   if (!remote)
     {
       g_dbus_method_invocation_return_error (invocation,
